File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini=prices[0]
        maxprofit=0
        for i in prices:
            cost=i-mini
            maxprofit=max(maxprofit,cost)
            mini=min(mini,i)
        return maxprofit    

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int min = INT_MAX;
        int maxprofit = 0;
        int profit = 0;
        for(int i = 0; i < n; i++) {
            if(min > prices[i]) {
                min = prices[i];
            }
            profit = prices[i] - min;
            if(profit > maxprofit) {
                maxprofit = profit;
            }
        }
        return maxprofit;
    }
};